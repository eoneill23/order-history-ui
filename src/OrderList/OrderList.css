.order-section {
  background-color: gray;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
}