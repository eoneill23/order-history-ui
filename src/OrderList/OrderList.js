import Order from '../Order/Order.js';
import './OrderList.css';
import React from 'react';

const OrderList = ({ orders }) => {
  let listOfOrders = orders.map(order => {
    return <Order key={order.id} id={order.id} img={order.img} name={order.name} description={order.description} price={order.price}/>
  })

  return (
    <section className='order-section'>
      {listOfOrders}
    </section>
  )
}

export default OrderList;