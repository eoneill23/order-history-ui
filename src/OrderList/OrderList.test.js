import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import OrderList from './OrderList.js';

describe('OrderList', () => {
  let wrapper, deleteOrderMock;

  beforeEach(() => {
    deleteOrderMock = jest.fn();
    wrapper = shallow(<OrderList orders={
      [
        { id: 1, img: 'google.com', name: 'Television', price: '200' },
        { id: 2, img: 'facebook.com', name: 'Radio', price: '100' }
      ]
    }
    deleteOrder={deleteOrderMock}/>)
  });

  it('should render without crashing', () => {
    expect(wrapper.length).toEqual(1)
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});