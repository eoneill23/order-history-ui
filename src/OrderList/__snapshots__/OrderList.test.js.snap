// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderList should match the snapshot with all data passed in correctly 1`] = `
<section
  className="order-section"
>
  <Order
    deleteOrder={[MockFunction]}
    id={1}
    img="google.com"
    key="1"
    name="Television"
    price="200"
  />
  <Order
    deleteOrder={[MockFunction]}
    id={2}
    img="facebook.com"
    key="2"
    name="Radio"
    price="100"
  />
</section>
`;
